lib/common/notifications.js

```diff
@@ -17,6 +17,22 @@ var _validateDocument = function(notification) {
     badge: Match.Optional(Number),
     sound: Match.Optional(String),
     notId: Match.Optional(Match.Integer),
+    apn: Match.Optional({
+      from: Match.Optional(String),
+      title: Match.Optional(String),
+      text: Match.Optional(String),
+      badge: Match.Optional(Number),
+      sound: Match.Optional(String),
+      notId: Match.Optional(Match.Integer)
+    }),
+    gcm: Match.Optional({
+      from: Match.Optional(String),
+      title: Match.Optional(String),
+      text: Match.Optional(String),
+      badge: Match.Optional(Number),
+      sound: Match.Optional(String),
+      notId: Match.Optional(Match.Integer)
+    }),
```

Comment:
We prop. could make this DRY
eg.:
```js
  var messageType = {
    from: Match.Optional(String),
    title: Match.Optional(String),
    text: Match.Optional(String),
    badge: Match.Optional(Number),
    sound: Match.Optional(String),
    notId: Match.Optional(Match.Integer)
  };

  var notificationType = _.default({
    from: String,
    title: String,
    text: String,
    apn: Match.Optional(messageType),
    gcm: Match.Optional(messageType),
    query: Match.Optional(String),
    token: Match.Optional(_matchToken),
    tokens: Match.Optional([_matchToken]),
    payload: Match.Optional(Object),
    delayUntil: Match.Optional(Date),
    createdAt: Date,
    createdBy: Match.OneOf(String, null)
  }, messageType);

  // Check the general notification
  check(notification, notificationType);
```

But it might be a bad idea since we loose some of the readability, opinions?